lib/travis/build/script/csharp.rb

```diff
@@ -48,17 +62,38 @@ def export
         end
 
         def install
-          sh.cmd "nuget restore #{config[:solution]}", retry: true if config[:solution]
+          sh.cmd "nuget restore #{config[:solution]}", retry: true if config[:solution] && !is_mono_2?
         end
 
         def script
           if config[:solution]
-            sh.cmd "xbuild /p:Configuration=Release #{config[:solution]}", timing: true if config[:solution]
+            sh.cmd "xbuild /p:Configuration=Release #{config[:solution]}", timing: true
           else
             sh.echo 'No solution or script defined, exiting', ansi: :red
             sh.cmd 'false', echo: false, timing: false
           end
         end
+
+        def mono_repo
+          if config[:mono] == 'latest'
+            'wheezy'
+          else
+            'wheezy/snapshots/' << config[:mono]
+          end
+        end
+
+        def mono_version_valid?
+          return true if config[:mono] == 'latest'
+          return false unless MONO_VERSION_REGEXP === config[:mono]
+
+          return false if MONO_VERSION_REGEXP.match(config[:mono])[1] == '2' && !is_mono_2?
```

Comment:
Yeah of course :) I just figured we can catch some dumb mistakes. Can we check that major >= 2?